### PR TITLE
Wash small critters in sinks

### PR DESCRIPTION
**Pull request by TDHooligan**
[CRITTER][FEATURE]

## About the PR 
You can place small animal critters in the sink & wash them with an advanced grab. cats are displeased by this.

https://github.com/goonstation/goonstation/assets/6061314/c5327de4-08b9-4eb3-ae9f-eb7a35d5532b

## Why's this needed? 
it was requested of me. i coded it as it's silly/cute/devious for rp. like washing remy before cutting them up

### CODE REVIEW

File: code/obj/submachine/cooking.dm [hunk 1]
```diff
@@ -42,8 +42,23 @@ TYPEINFO(/obj/submachine/chef_sink)
...
+			var/obj/item/grab/GRAB = W
+			if (ismob(GRAB.affecting))
+				if (GRAB.state >= 1 && istype(GRAB.affecting, /mob/living/critter/small_animal))
+					var/mob/M = GRAB.affecting
+					var/mob/A = GRAB.assailant
+					if (BOUNDS_DIST(src.loc, M.loc) > 0)
+						return
+					user.visible_message("<span class='notice'>[A] shoves [M] in the sink and starts to wash them.</span>")
+					M.set_loc(src.loc)
+					playsound(src.loc, 'sound/impact_sounds/Liquid_Slosh_1.ogg', 25, 1)
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:54]: I lowered the volume of hand washing a bit yesterday. Mirroring here.
```suggestion
					playsound(src.loc, 'sound/impact_sounds/Liquid_Slosh_1.ogg', 15, 1)
```

File: code/obj/submachine/cooking.dm [hunk 2]
```diff
@@ -42,8 +42,23 @@ TYPEINFO(/obj/submachine/chef_sink)
...
+					actions.start(new/datum/action/bar/private/handwashing/critterwashing(A,src,M,GRAB),user)
+				else
+					playsound(src.loc, 'sound/impact_sounds/Liquid_Slosh_1.ogg', 25, 1)
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:57]:
```suggestion
					playsound(src.loc, 'sound/impact_sounds/Liquid_Slosh_1.ogg', 15, 1)
```

File: code/obj/submachine/cooking.dm [hunk 3]
```diff
@@ -146,6 +161,56 @@ TYPEINFO(/obj/submachine/chef_sink)
 		..()
 
 
+/datum/action/bar/private/handwashing/critterwashing
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:164]: What behaviour does this inherit from the handwashing action? Is this inheritnce necessary?

File: code/obj/submachine/cooking.dm [hunk 4]
```diff
@@ -146,6 +161,56 @@ TYPEINFO(/obj/submachine/chef_sink)
...
+		..()
+
+	checkStillValid()
+		if(GET_DIST(victim, sink) > 0 || BOUNDS_DIST(user, sink) > 1 || victim == null || user == null || sink == null || !grab)
+			interrupt(INTERRUPT_ALWAYS)
+			return FALSE
+		return TRUE
+	onStart()
+		if (istype(victim, /mob/living/critter/small_animal/cat) && victim.ai?.enabled)
+			victim._ai_patience_count = 0
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:183]: The leading underscore feels like this is meant to be a private variable to me. Is it really ok to modify it like this?

Comment (TDHooligan) [reply code/obj/submachine/cooking.dm:183]: it's not great but there's really no other way i could spot, i could make a setter function if that's a better looking outcome

File: code/obj/submachine/cooking.dm (hunk 3 above)
Comment (TDHooligan) [reply code/obj/submachine/cooking.dm:164]: it originally had more in common, but grew in scope when critter ai got involved. on it

File: code/obj/submachine/cooking.dm [hunk 5]
```diff
@@ -146,6 +161,61 @@ TYPEINFO(/obj/submachine/chef_sink)
...
+			victim.was_harmed(user)
+			victim.visible_message("<span class='notice'>[victim] resists [user]'s attempt to wash them!</span>")
+			playsound(victim.loc, 'sound/voice/animal/cat_hiss.ogg', 50, 1)
+
+		else if (victim.ai?.enabled && istype(victim.ai.current_task, /datum/aiTask/timed/wander) )
+			victim.ai.wait(5)
+		..()
+	loopStart()
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:195]:
```suggestion

	loopStart()
```

File: code/obj/submachine/cooking.dm [hunk 6]
```diff
@@ -146,6 +161,62 @@ TYPEINFO(/obj/submachine/chef_sink)
...
+		if(!checkStillValid())
+			..()
+			return
+		victim.blood_DNA = null
+		victim.blood_type = null
+		victim.forensics_blood_color = null
+		victim.set_clothing_icon_dirty()
+
+		src.onRestart()
+	onInterrupt()
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:217]:
```suggestion

	onInterrupt()
```

File: code/obj/submachine/cooking.dm [hunk 7]
```diff
@@ -146,6 +161,62 @@ TYPEINFO(/obj/submachine/chef_sink)
...
+		..()
```
Comment (pali6) [inline code/obj/submachine/cooking.dm:218]:
```suggestion
```

**Review by pali6 — APPROVED** [review]

**Pull request merged**